### ISSUE TITLE
typeinfo hooks raising exceptions

### ISSUE THREAD

**Issue by c0d3z3r0** [arizvisa/ida-minsc#141]
Getting these exceptions when using `ida-efitools2` with `ida-minsc` loaded. Commenting out that logging line, other ones in hooks.py fail, too. IDA Pro 7.6.210427

~~~
CRITICAL:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) : Callback for IDB_Hooks.changing_ti with priority (+0) raised an exception while executing <bound method type.changing of <class 'hooks.typeinfo'>>
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) : Traceback (IDB_Hooks.changing_ti was hooked at)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :   File "C:\Program Files\IDA7.5.20.1028\python\2\ida_idp.py", line 6066, in dispatch
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :     cb.fun(slot, *args)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\base\_interface.py", line 524, in closure
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :     result = callable(*parameters)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\misc\hooks.py", line 1486, in make_ida_not_suck_cocks
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :     ui.hook.idb.add('changing_ti', typeinfo.changing, 0)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) : 
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) : Traceback (most recent call last):
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\base\_interface.py", line 524, in closure
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :     result = callable(*parameters)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\misc\hooks.py", line 644, in changing
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :     return logging.debug(u"{:s}.changing({:#x}, {!s}, {!s}) : Ignoring typeinfo.changing event (not an address) with new type ({!s}) and new name ({!s}) at {:#x}.".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(new_type), utils.string.repr(new_fname), new_type, new_fname, ea))
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) : UnicodeDecodeError: 'ascii' codec can't decode byte 0x8e in position 4: ordinal not in range(128)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) : 
CRITICAL:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) : Callback for IDB_Hooks.ti_changed with priority (+0) raised an exception while executing <bound method type.changed of <class 'hooks.typeinfo'>>
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) : Traceback (IDB_Hooks.ti_changed was hooked at)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :   File "C:\Program Files\IDA7.5.20.1028\python\2\ida_idp.py", line 6066, in dispatch
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :     cb.fun(slot, *args)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\base\_interface.py", line 524, in closure
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :     result = callable(*parameters)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\misc\hooks.py", line 1487, in make_ida_not_suck_cocks
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :     ui.hook.idb.add('ti_changed', typeinfo.changed, 0)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) : 
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) : Traceback (most recent call last):
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\base\_interface.py", line 524, in closure
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :     result = callable(*parameters)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\misc\hooks.py", line 692, in changed
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) :     return logging.debug(u"{:s}.changed({:#x}, {!s}, {!s}) : Ignoring typeinfo.changed event (not an address) with type ({!s}) and name ({!s}) at {:#x}.".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(type), utils.string.repr(fnames), type, fnames, ea))
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) : UnicodeDecodeError: 'ascii' codec can't decode byte 0x8e in position 4: ordinal not in range(128)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107521L, '\n=\x04#\x8eu', None) : 
CRITICAL:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) : Callback for IDB_Hooks.changing_ti with priority (+0) raised an exception while executing <bound method type.changing of <class 'hooks.typeinfo'>>
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) : Traceback (IDB_Hooks.changing_ti was hooked at)
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :   File "C:\Program Files\IDA7.5.20.1028\python\2\ida_idp.py", line 6066, in dispatch
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :     cb.fun(slot, *args)
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\base\_interface.py", line 524, in closure
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :     result = callable(*parameters)
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\misc\hooks.py", line 1486, in make_ida_not_suck_cocks
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :     ui.hook.idb.add('changing_ti', typeinfo.changing, 0)
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) : 
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) : Traceback (most recent call last):
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\base\_interface.py", line 524, in closure
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :     result = callable(*parameters)
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\misc\hooks.py", line 656, in changing
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :     logging.debug(u"{:s}.changing({:#x}, {!s}, {!s}) : Received typeinfo.changing for new_type ({!s}) and new_fname ({!s}).".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(new_type), utils.string.repr(new_fname), new_type, new_fname))
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) : UnicodeDecodeError: 'ascii' codec can't decode byte 0x83 in position 4: ordinal not in range(128)
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) : 
CRITICAL:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) : Callback for IDB_Hooks.ti_changed with priority (+0) raised an exception while executing <bound method type.changed of <class 'hooks.typeinfo'>>
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) : Traceback (IDB_Hooks.ti_changed was hooked at)
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :   File "C:\Program Files\IDA7.5.20.1028\python\2\ida_idp.py", line 6066, in dispatch
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :     cb.fun(slot, *args)
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\base\_interface.py", line 524, in closure
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :     result = callable(*parameters)
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\misc\hooks.py", line 1487, in make_ida_not_suck_cocks
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :     ui.hook.idb.add('ti_changed', typeinfo.changed, 0)
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) : 
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) : Traceback (most recent call last):
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\base\_interface.py", line 524, in closure
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :     result = callable(*parameters)
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\misc\hooks.py", line 705, in changed
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) :     logging.debug(u"{:s}.changed({:#x}, {!s}, {!s}) : Received typeinfo.changed event with type ({!s}) and name ({!s}).".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(type), utils.string.repr(fnames), type, fnames))
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) : UnicodeDecodeError: 'ascii' codec can't decode byte 0x83 in position 4: ordinal not in range(128)
WARNING:root:internal.interface.priorityhook.callback(2147495704L, '\n=\x05#\x83\x85f', None) : 
CRITICAL:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) : Callback for IDB_Hooks.changing_ti with priority (+0) raised an exception while executing <bound method type.changing of <class 'hooks.typeinfo'>>
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) : Traceback (IDB_Hooks.changing_ti was hooked at)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :   File "C:\Program Files\IDA7.5.20.1028\python\2\ida_idp.py", line 6066, in dispatch
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :     cb.fun(slot, *args)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\base\_interface.py", line 524, in closure
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :     result = callable(*parameters)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\misc\hooks.py", line 1486, in make_ida_not_suck_cocks
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :     ui.hook.idb.add('changing_ti', typeinfo.changing, 0)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) : 
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) : Traceback (most recent call last):
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\base\_interface.py", line 524, in closure
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :     result = callable(*parameters)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\misc\hooks.py", line 644, in changing
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :     return logging.debug(u"{:s}.changing({:#x}, {!s}, {!s}) : Ignoring typeinfo.changing event (not an address) with new type ({!s}) and new name ({!s}) at {:#x}.".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(new_type), utils.string.repr(new_fname), new_type, new_fname, ea))
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) : UnicodeDecodeError: 'ascii' codec can't decode byte 0x83 in position 4: ordinal not in range(128)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) : 
CRITICAL:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) : Callback for IDB_Hooks.ti_changed with priority (+0) raised an exception while executing <bound method type.changed of <class 'hooks.typeinfo'>>
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) : Traceback (IDB_Hooks.ti_changed was hooked at)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :   File "C:\Program Files\IDA7.5.20.1028\python\2\ida_idp.py", line 6066, in dispatch
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :     cb.fun(slot, *args)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\base\_interface.py", line 524, in closure
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :     result = callable(*parameters)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\misc\hooks.py", line 1487, in make_ida_not_suck_cocks
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :     ui.hook.idb.add('ti_changed', typeinfo.changed, 0)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) : 
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) : Traceback (most recent call last):
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\base\_interface.py", line 524, in closure
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :     result = callable(*parameters)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\misc\hooks.py", line 692, in changed
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) :     return logging.debug(u"{:s}.changed({:#x}, {!s}, {!s}) : Ignoring typeinfo.changed event (not an address) with type ({!s}) and name ({!s}) at {:#x}.".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(type), utils.string.repr(fnames), type, fnames, ea))
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) : UnicodeDecodeError: 'ascii' codec can't decode byte 0x83 in position 4: ordinal not in range(128)
WARNING:root:internal.interface.priorityhook.callback(18374686481819107530L, '\n=\x05#\x83\x85g', None) : 
~~~

**Comment by arizvisa** [comment]
Okay. It looks like I'm not escaping the type information that should be displayed. This typeinfo hook is so that you can query any types that you apply to a function prototype via `database.select` using the `__typeinfo__` key, and that way you can use it to figure out what functions you've visited after a database has been processed. When you commented out those logging lines, what was the other error that you were getting?

I'll try and fix the logging error (might happen sometime tomorrow), and then we can look at what the second error that you're getting is.

**Comment by arizvisa** [comment]
@c0d3z3r0, you still around for this bug report?

**Comment by c0d3z3r0** [comment]
Thanks for the reminder!

Seems like that does the job. The other errors are the same, just different log messages.

This patch solved all, however, I'm unsure if wrapping `fname*` is needed or not, since I can't trigger any errors with that parameter.

~~~
diff --git a/misc/hooks.py b/misc/hooks.py
index e6760be..3443687 100644
--- a/misc/hooks.py
+++ b/misc/hooks.py
@@ -639,21 +639,21 @@ class typeinfo(changingchanged):
     @classmethod
     def changing(cls, ea, new_type, new_fname):
         if not cls.is_ready():
-            return logging.debug(u"{:s}.changing({:#x}, {!s}, {!s}) : Ignoring typeinfo.changing event (database not ready) with new type ({!s}) and new name ({!s}) at {:#x}.".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(new_type), utils.string.repr(new_fname), new_type, new_fname, ea))
+            return logging.debug(u"{:s}.changing({:#x}, {!s}, {!s}) : Ignoring typeinfo.changing event (database not ready) with new type ({!s}) and new name ({!s}) at {:#x}.".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(new_type), utils.string.repr(new_fname), utils.string.repr(new_type), utils.string.repr(new_fname), ea))
         if interface.node.is_identifier(ea):
-            return logging.debug(u"{:s}.changing({:#x}, {!s}, {!s}) : Ignoring typeinfo.changing event (not an address) with new type ({!s}) and new name ({!s}) at {:#x}.".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(new_type), utils.string.repr(new_fname), new_type, new_fname, ea))
+            return logging.debug(u"{:s}.changing({:#x}, {!s}, {!s}) : Ignoring typeinfo.changing event (not an address) with new type ({!s}) and new name ({!s}) at {:#x}.".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(new_type), utils.string.repr(new_fname), utils.string.repr(new_type), utils.string.repr(new_fname), ea))
 
         # Verify that the address is within our database boundaries because IDA
         # can actually create "extra" comments outside of the database.
         try:
             ea = interface.address.within(ea)
         except E.OutOfBoundsError:
-            return logging.debug(u"{:s}.changing({:#x}, {!s}, {!s}) : Ignoring typeinfo.changing event (not a valid address) with new type ({!s}) and new name ({!s}) at {:#x}.".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(new_type), utils.string.repr(new_fname), new_type, new_fname, ea))
+            return logging.debug(u"{:s}.changing({:#x}, {!s}, {!s}) : Ignoring typeinfo.changing event (not a valid address) with new type ({!s}) and new name ({!s}) at {:#x}.".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(new_type), utils.string.repr(new_fname), utils.string.repr(new_type), utils.string.repr(new_fname), ea))
 
         # Extract the previous type information from the given address. If none
         # was found, then just use empty strings because these are compared to the
         # new values by the event.
-        logging.debug(u"{:s}.changing({:#x}, {!s}, {!s}) : Received typeinfo.changing for new_type ({!s}) and new_fname ({!s}).".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(new_type), utils.string.repr(new_fname), new_type, new_fname))
+        logging.debug(u"{:s}.changing({:#x}, {!s}, {!s}) : Received typeinfo.changing for new_type ({!s}) and new_fname ({!s}).".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(new_type), utils.string.repr(new_fname), utils.string.repr(new_type), utils.string.repr(new_fname)))
 
         ti = database.type(ea)
         old_type, old_fname, _ = (b'', b'', None) if ti is None else ti.serialize()
@@ -678,6 +678,7 @@ class typeinfo(changingchanged):
         # coroutine has gone out of control and we need to let the user know.
         except StopIteration as E:
             logging.fatal(u"{:s}.changed({:#x}, {!s}, {!s}) : Abandoning type information at {:#x} due to unexpected termination of event handler.".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(new_type), utils.string.repr(new_fname), ea), exc_info=True)
+            pass
 
         # Last thing to do is to re-enable the hooks that we disabled and then leave.
         finally:
@@ -687,22 +688,22 @@ class typeinfo(changingchanged):
     @classmethod
     def changed(cls, ea, type, fnames):
         if not cls.is_ready():
-            return logging.debug(u"{:s}.changed({:#x}, {!s}, {!s}) : Ignoring typeinfo.changed event (database not ready) with type ({!s}) and name ({!s}) at {:#x}.".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(type), utils.string.repr(fnames), type, fnames, ea))
+            return logging.debug(u"{:s}.changed({:#x}, {!s}, {!s}) : Ignoring typeinfo.changed event (database not ready) with type ({!s}) and name ({!s}) at {:#x}.".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(type), utils.string.repr(fnames), utils.string.repr(type), fnames, ea))
         if interface.node.is_identifier(ea):
-            return logging.debug(u"{:s}.changed({:#x}, {!s}, {!s}) : Ignoring typeinfo.changed event (not an address) with type ({!s}) and name ({!s}) at {:#x}.".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(type), utils.string.repr(fnames), type, fnames, ea))
+            return logging.debug(u"{:s}.changed({:#x}, {!s}, {!s}) : Ignoring typeinfo.changed event (not an address) with type ({!s}) and name ({!s}) at {:#x}.".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(type), utils.string.repr(fnames), utils.string.repr(type), fnames, ea))
 
         # Verify that the address is within our database boundaries because IDA
         # can actually create "extra" comments outside of the database.
         try:
             ea = interface.address.within(ea)
         except E.OutOfBoundsError:
-            return logging.debug(u"{:s}.changed({:#x}, {!s}, {!s}) : Ignoring typeinfo.changed event (not a valid address) with type ({!s}) and name ({!s}) at {:#x}.".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(type), utils.string.repr(fnames), type, fnames, ea))
+            return logging.debug(u"{:s}.changed({:#x}, {!s}, {!s}) : Ignoring typeinfo.changed event (not a valid address) with type ({!s}) and name ({!s}) at {:#x}.".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(type), utils.string.repr(fnames), utils.string.repr(type), fnames, ea))
 
         # Resume the state for the current address, and then take the data from
         # our parameters (which IDA is telling us was just written) and pack
         # them into a tuple. This way we can send them to the state after we
         # disable the necessary hooks to prevent re-entrancy.
-        logging.debug(u"{:s}.changed({:#x}, {!s}, {!s}) : Received typeinfo.changed event with type ({!s}) and name ({!s}).".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(type), utils.string.repr(fnames), type, fnames))
+        logging.debug(u"{:s}.changed({:#x}, {!s}, {!s}) : Received typeinfo.changed event with type ({!s}) and name ({!s}).".format('.'.join([__name__, cls.__name__]), ea, utils.string.repr(type), utils.string.repr(fnames), utils.string.repr(type), fnames))
         event, new = cls.resume(ea), (type or b'', fnames or b'')
 
         # First disable our hooks so that we can prevent re-entrancy issues.
~~~

**Comment by arizvisa** [comment]
okay, so there's no other issues, then?

**Comment by c0d3z3r0** [comment]
> okay, so there's no other issues, then?

right, with the patch above there are no more issues

**Comment by arizvisa** [comment]
Okay, I'll write up a PR. Thanks for your report, good sir.

**Comment by arizvisa** [comment]
This issue was fixed by PR #142 which has been merged.

Thus, this issue is being closed. If you feel this was in error and the issue still exists, please let me know and I can re-open the ticket.
